pkg/actions/addon/tasks.go

```diff
@@ -178,15 +178,11 @@ func createAddonManager(ctx context.Context, clusterProvider *eks.ClusterProvide
 		oidcProviderExists bool
 	)
 	if api.IsEnabled(cfg.IAM.WithOIDC) {
-		var err error
-		oidc, err = clusterProvider.NewOpenIDConnectManager(ctx, cfg)
-		if err != nil {
-			return nil, err
-		}
-		oidcProviderExists, err = oidc.CheckProviderExists(ctx)
-		if err != nil {
-			return nil, err
-		}
+		// We are here because we are creating a cluster with OIDC enabled.
+		// To avoid a race, assume that it is getting created properly
+		// and do not check explicitly if it is already ready. If we really
+		// want to be sure then we would have to loop/wait/timeout here.
+		oidcProviderExists = true
```

Comment:
I'm not great with go - but is this just simply:

```go
  oidcProviderExists := api.IsEnabled(cfg.IAM.WithOIDC)
```

removing the variable initialization above

Reply:
Sorry should have been a draft :)